fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
+            onBackButtonClick = {},
+            onBackButtonLongPress = {},
+        )
+
+        ForwardButton(
+            onForwardButtonClick = {},
+            onForwardButtonLongPress = {},
+        )
+
+        SearchWebButton(
+            onSearchButtonClick = onSearchButtonClick,
+        )
+
+        TabsButton(
+            onTabsButtonClick = onTabsButtonClick,
+            tabCount = tabCount,
+        )
+
+        MenuButton(menuButton = menuButton)
+    }
 }
 
-/**
- * Provides a collection of navigation items used in the application's navigation bar.
- */
-object NavigationItems {
-    val home = ActionItem(
-        iconId = R.drawable.mozac_ic_home_24,
-        descriptionResourceId = R.string.browser_toolbar_home,
-    )
+@Composable
+private fun BackButton(
+    onBackButtonClick: () -> Unit,
+    onBackButtonLongPress: () -> Unit,
+) {
+    LongPressIconButton(
+        onClick = { onBackButtonClick() },
+        onLongClick = { onBackButtonLongPress() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_back_24),
+            stringResource(id = R.string.browser_menu_back),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
 
-    val menu = ActionItem(
-        iconId = R.drawable.mozac_ic_ellipsis_vertical_24,
-        descriptionResourceId = R.string.mozac_browser_menu_button,
-        type = ItemType.MENU,
-    )
+@Composable
+private fun ForwardButton(
+    onForwardButtonClick: () -> Unit,
+    onForwardButtonLongPress: () -> Unit,
+) {
+    LongPressIconButton(
+        onClick = { onForwardButtonClick() },
+        onLongClick = { onForwardButtonLongPress() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_forward_24),
+            stringResource(id = R.string.browser_menu_forward),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
 
-    val back = ActionItem(
-        iconId = R.drawable.mozac_ic_back_24,
-        descriptionResourceId = R.string.browser_menu_back,
+@Composable
+private fun HomeButton(
+    onHomeButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onHomeButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_home_24),
+            stringResource(id = R.string.browser_toolbar_home),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun SearchWebButton(
+    onSearchButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onSearchButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_search_24),
+            stringResource(id = R.string.search_hint),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun MenuButton(
+    menuButton: MenuButton,
+) {
+    AndroidView(
+        modifier = Modifier.size(48.dp),
+        factory = { _ -> menuButton },
     )
+}
```

Comment:
Is there some special business logic with `MenuButton` that we want to keep by implementing an `AndroidView` here? I'm curious if we'd be better off just having another `IconButton` here and passing up the click lambda to the fragments, so they can open the menu.

Reply:
menu building and displaying logic is [encapsulated](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/view/MenuButton.kt#99) in the menubutton class itself, refactoring it will take a bit of extra work. Also, the menu itself will be refactored after the toolbar redesign, let's refactor it after we get the new menu
filled up a [follow up](https://bugzilla.mozilla.org/show_bug.cgi?id=1884049)